nevergrad/optimization/optimizerlib.py

```diff
@@ -78,10 +80,16 @@ def __init__(
                 assert isinstance(noise_handling, tuple), "noise_handling must be a string or  a tuple of type (strategy, factor)"
                 assert noise_handling[1] > 0.0, "the factor must be a float greater than 0"
                 assert noise_handling[0] in ["random", "optimistic"], f"Unkwnown noise handling: '{noise_handling}'"
-        assert mutation in ["gaussian", "cauchy", "discrete", "fastga", "doublefastga", "adaptive",
+        assert mutation in ["gaussian", "cauchy", "discrete", "fastga", "doublefastga", "adaptive", "coordinatewiseadaptive",
                             "portfolio", "discreteBSO", "lengler", "doerr"], f"Unkwnown mutation: '{mutation}'"
         if mutation == "adaptive":
             self._adaptive_mr = 0.5
+        if mutation == "coordinatewiseadaptive":
+            all_params = paramhelpers.flatten_parameter(self.parametrization)
+            arity = max(len(param.choices) if isinstance(param, p.TransitionChoice) else 500 for param in all_params.values())
```

Comment:
why is that duplicated from line 71? please remove if not needed

And I am not sure I like the 500 value, is there any reason for using this? this seems artificial

Reply:
An educated guess could be like self.budget ** (1/3)    (inspired by progressive widening). However the fact that this is coordinate-wise makes the guess difficult.
This is "how much do we discretize in continuous domains".

Reply:
Duplicate code removed.